controllers/watcherapi_controller.go

```diff
@@ -181,25 +211,65 @@ func (r *WatcherAPIReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 }
 
 // generateServiceConfigs - create Secret which holds the service configuration
-// NOTE - jgilaber this function is WIP, currently implements a fraction of its
-// functionality and will be expanded of further iteration to actually generate
-// the service configs
 func (r *WatcherAPIReconciler) generateServiceConfigs(
 	ctx context.Context, instance *watcherv1beta1.WatcherAPI,
 	secret corev1.Secret, db *mariadbv1.Database,
+	memcachedInstance *memcachedv1.Memcached,
 	helper *helper.Helper, envVars *map[string]env.Setter,
 ) error {
 	Log := r.GetLogger(ctx)
 	Log.Info("generateServiceConfigs - reconciling")
 
-	// replace by actual usage in future iterations
-	_ = db
-	_ = helper
-	_ = instance
-	_ = secret
-	_ = envVars
+	labels := labels.GetLabels(instance, labels.GetGroupLabel(watcher.ServiceName), map[string]string{})
+	// jgilaber this might be wrong? we should probably get keystonapi in the
+	// watcher controller and set the url in the spec eventually?
+	keystoneAPI, err := keystonev1.GetKeystoneAPI(ctx, helper, instance.Namespace, map[string]string{})
+	// KeystoneAPI not available we should not aggregate the error and continue
+	if err != nil {
+		instance.Status.Conditions.Set(condition.FalseCondition(
+			condition.ServiceConfigReadyCondition,
+			condition.ErrorReason,
+			condition.SeverityWarning,
+			condition.ServiceConfigReadyErrorMessage,
+			"keystoneAPI not found"))
+		return err
+	}
+	keystoneInternalURL, err := keystoneAPI.GetEndpoint(endpoint.EndpointInternal)
+	if err != nil {
+		return err
+	}
+	// customData hold any customization for the service.
+	// NOTE jgilaber making an empty map for now, we'll probably want to
+	// implement CustomServiceConfig later
+	customData := map[string]string{}
+
+	databaseAccount := db.GetAccount()
+	databaseSecret := db.GetSecret()
+	templateParameters := map[string]interface{}{
+		"DatabaseConnection": fmt.Sprintf("mysql+pymysql://%s:%s@%s/%s?charset=utf8&plugin=dbcounter",
```

Comment:
I had errors in the service when setting plugin=dbcounter. It mentioned something like dbcounter plugin not available or similar error. It's something specific to devstack installation. I see other services are adding `read_default_file=/etc/my.cnf` . I think it's used in some cases (TLS?)

Reply:
I used a devstack config as a starting point, so it makes sense that it's devstack related. I haven't really been able to test this, so let me remove this value from the config, and I'll tune the config once I'm able to create a deployment